controllers/dependencies.go

```diff
@@ -0,0 +1,167 @@
+package controllers
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+
+	"os"
+
+	"path/filepath"
+
+	git "github.com/go-git/go-git/v5"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	OPERATOR_PIPELINES_REPO = "https://github.com/redhat-openshift-ecosystem/operator-pipelines.git"
+	REPO_CLONE_PATH         = "/tmp/operator-pipelines/"
+	PIPELINE_MANIFESTS_PATH = "ansible/roles/operator-pipeline/templates/openshift/pipelines"
+	TASKS_MANIFESTS_PATH    = "ansible/roles/operator-pipeline/templates/openshift/tasks"
+)
+
+func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.ObjectMeta) error {
+
+	// Cloning operator-pipelines project to retrieve pipelines and tasks
+	// yaml manifests that need to be applied beforehand
+	// ref: https://github.com/redhat-openshift-ecosystem/certification-releases/blob/main/4.9/ga/ci-pipeline.md#step-6---install-the-certification-pipeline-and-dependencies-into-the-cluster
+
+	_, err := git.PlainClone(REPO_CLONE_PATH, false, &git.CloneOptions{
+		URL:      OPERATOR_PIPELINES_REPO,
+		Progress: os.Stdout,
+	})
+	if err != nil {
+		log.Log.Info("Couldn't clone the repository for operator-pipelines.")
+		return err
+	}
+
+	// Reading pipeline manifests and applying to cluster
+	root := REPO_CLONE_PATH + PIPELINE_MANIFESTS_PATH
+	err = filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			if errors := r.applyManifests(path, meta.Namespace); errors != nil {
+				return errors
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		log.Log.Info("Couldn't iterate over operator-pipelines yaml manifest files")
+		return err
+	}
+
+	// Reading tasks manifests and applying it to cluster
+	root = REPO_CLONE_PATH + TASKS_MANIFESTS_PATH
+	err = filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			if errors := r.applyManifests(path, meta.Namespace); errors != nil {
+				return errors
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		log.Log.Info("Couldn't iterate over operator-pipelines yaml manifest files")
+		return err
+	}
+
+	// Removing cloned project
+	if err = os.RemoveAll(PIPELINE_MANIFESTS_PATH); err != nil {
+		log.Log.Info("Couldn't remove operator-pipelines directory")
+		return err
+	}
+	return nil
+}
+
+func (r *OperatorPipelineReconciler) applyManifests(fileName string, Namespace string) error {
+
+	b, err := ioutil.ReadFile(fileName)
```

Comment:
ioutil is going to be deprecated soon (I think). It would be better to use `os` package instead.

Reply:
Not sure about ioutil. But no problem os is now the one on the code. They have exactly the same signature.